codegen/src/main/scala/caliban/codegen/CodegenPlugin.scala

```diff
@@ -57,23 +60,31 @@ object CodegenPlugin extends AutoPlugin {
       case _                                         => putStrLn(helpMsg)
     }
 
+  def getSchema(path: String): Task[Document] =
```

Comment:
Depending on their configuration they may need to pass additional headers when introspecting, should we include that?

Reply:
I won't do it right now but I agree. The current code for parsing options isn't very convenient, maybe I should use something like https://github.com/scopt/scopt. Will make an issue.